.gitlab-ci.yml

```diff
@@ -0,0 +1,72 @@
+---
+
+image: quay.io/govcms/govcms-ci
+
+services:
+  - "docker:dind"
+
+stages:
+  - build
+
+build:
+
+  stage: build
+
+  artifacts:
+    paths:
+      - $CI_PROJECT_DIR/logs
+
+  script:
+
+    # Export dynamic variables
+    - export DOCKERHUB_NAMESPACE="${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}"
+    - export IMAGE_VERSION_TAG_PREFIX="";
+
+    # Handle master branch for IMAGE_VERSION_TAG gracefully.
+    - if [[ "${CI_COMMIT_REF_NAME}" == "master" ]]; then
+        export IMAGE_VERSION_TAG="latest";
```

Comment:
Could this just literally be "master", which then leaves :latest as a magic docker thing we don't use?

Reply:
Well, we could use latest as a default however this is more explicit.

You will need to test this within the govCMS8lagoon repository mirrored in the GovCMS gitlab instance. You currently have access to this one - please refer to the branch `feature/gitlab-ci-extended`.

This is because there are variable references which refer to specific authentication configuration of the docker registry that cannot be publicly exposed.

Reply:
> Well, we could use latest as a default however this is more explicit.

No i mean use "master" as default and literally never ever refer to "latest".

I will try to test.

Reply:
Looks good to me
```

-----File-----

These entries have been added to registry.hub.docker.com/govcms8lagoon/chrome:latest: None

These entries have been deleted from registry.hub.docker.com/govcms8lagoon/chrome:latest: None

These entries have been changed between registry.hub.docker.com/govcms8lagoon/chrome:latest and gitlab-registry-production.govcms.amazee.io/dof-dev/govcms8lagoon/chrome:feature-gitlab-ci-extended: None
```

This is expected ^^ I guess because you're not changing the images in these tests.

```
time="2019-09-09T02:24:26Z" level=error msg="Could not obtain size for /root/.container-diff/cache/REDACTED/usr/bin/bc: stat /root/.container-diff/cache/REDACTED/usr/bin/bc: no such file or directory"
```
Is this expected?  ^^

Reply:
It's expected - I'm not sure if there's a way to suppress them.